### DISCUSSION TITLE
Saving files to device on Thonny

### DISCUSSION THREAD

**Discussion by cecily555** [micropython/micropython#16203] [RP2040 / Pico] [answered]
I am working on a project to read pressure and temperature data from a sensor, saving it to a .csv file and exporting the data to another application to create a graph. I have written the following code in Thonny. I am using the Raspberry Pi Pico 2 as my microprocessor. 

`from machine import Pin, I2C
from utime import sleep
from bmp280 import BMP280I2C

i2c0_sda = Pin(0)
i2c0_scl = Pin(1)
i2c0 = I2C(0, sda=i2c0_sda, scl=i2c0_scl, freq=400000)
bmp280_i2c =BMP280I2C(0x77,i2c0) # address may be different
temp_and_pressure = open("temp_and_pressure_file.csv","a")

while True:
 readout = bmp280_i2c.measurements
 pressure = readout['p']
 temperature = readout['t']
 print("reading taken")
 try:
     temp_and_pressure.write(str(temperature))
     temp_and_pressure.write(",")
     temp_and_pressure.write(str(pressure))
     temp_and_pressure.write("\r\n")
 except OSError:
    print("Disk full?")
 sleep(1.5)`

However, the file is created and saved locally on the pico board, and I cannot work out how to graph this data. Is there any way to write code to create and save the file to my computer instead of the pico? Alternatively, could I directly export the file from my pico into a graphing application on my MacBook? I have already tried using matplotlib, but I am using MicroPython and can't install many of the modules it relies on (like abc). Can anyone help?

<img width="401" alt="image" src="https://github.com/user-attachments/assets/d8f1b852-786a-4bff-b668-8b281e27eec5">

**Comment by robert-hh** [comment, marked as answer]
You cannot write directly from the board into the PC's file system. It always needs some kind of agent to receive data sent by the board. A very simple method to do so is using `mpremote mount PC-path`, which map the PC directory `PC-path` as path `/remote` at the board. Data written to a file at `/remote` on the board is written to a file with the same name at the PC's `PC-path`. 
Other options would be to just print the data at the board and have a logger at the PC, which captures the data and write it to a file.
